docs/content/concepts/terminology.md

```diff
@@ -1,100 +1,142 @@
-multiple different models that result in logical clusters being created, with different policies or lifecycles, but
-Workspace is intended to be the most generic representation of the concept with the broadest possible utility to anyone
-building control planes.
+A workspace provides Kubernetes-cluster-like HTTPS/CRUD endpoints, i.e., endpoints that typical Kubernetes client
+tooling (e.g., client-go, controller-runtime, and others) can interact with as they would with classic Kubernetes
+clusters. Workspace provides multi-tenancy in a kcp setup.
+
+Each workspace is backed by a logical cluster, but not all logical clusters may be exposed as workspaces.
+Creating a `Workspace` object results in a logical cluster being created and available via an URL for the client to
+connect and create resources supported by the APIs in that workspace. The list of default APIs depends on the chosen
+Workspace type, which will be covered later in this document.
+
+There could be multiple different models that result in logical clusters being created, with different policies or
+lifecycle, but Workspace is intended to be the most generic representation of the concept with the broadest possible
+utility to anyone building control planes.
+
+A workspace can be used to:
+
+- export APIs, making them bindable inside different workspaces and logical clusters
+- bind APIs, i.e. "install" APIs from different workspaces and logical clusters into the current workspace and logical
+  cluster
+- allocate capacity for creating instances of those APIs (quota)
 
-A workspace binds APIs and makes them accessible inside the logical cluster, allocates capacity for creating instances
-of those APIs (quota), and defines how multi-workspace operations can be performed by users, clients, and controller
+Additionally, a workspace defines how multi-workspace operations can be performed by users, clients, and controller
 integrations.
 
-To a user, a workspace appears to be a Kubernetes cluster minus all the container orchestration specific resources. It
-has its own discovery, its own OpenAPI spec, and follows the kube-like constraints about uniqueness of
-Group-Version-Resource and its behaviour (no two GVRs with different schemas can exist per workspace, but workspaces can
-have different schemas). A user can define a workspace as a context in a kubeconfig file and `kubectl get all -A` would
-return all objects in all namespaces of that workspace.
+To a user, a workspace appears to be a Kubernetes cluster without all the container orchestration specific resources.
+(e.g. Pods, ReplicaSets, Deployments...). It has its own discovery, its own OpenAPI spec, and follows the
+Kubernetes-like constraints about uniqueness of Group-Version-Resource and its behavior. For example, it's not
+possible to have two identical GVRs with different schemas in one workspace, but it's possible to have two identical
+GVRs with different schemas in different workspaces.
```

Comment:
Wondering whether it is confusing that we describe all this for workspaces. It applies to logical clusters as well.

Reply:
To be very honest, I struggled with logical cluster vs workspaces in the very beginning, but it might be that I still struggle a little. I agree that this applies to logical clusters as well. A lot of stuff under workspaces and exporting/binding applies to both.

Based on some other comments and my current understanding, I'd describe Workspace as a way to create logical cluster with a predefined set of available APIs. In other words, it's an abstraction on top of logical clusters that's going to properly bootstrap and prepare a logical cluster for the use case that you want.

What I still don't understand is the point about CRUD endpoints. If that's what I get just with a logical cluster (which is my understanding of https://github.com/kcp-dev/kcp/pull/3238#discussion_r1898576991), what endpoints are actually provided by a Workspace?

Reply:
> Based on some other comments and my current understanding, I'd describe Workspace as a way to create logical cluster with a predefined set of available APIs. In other words, it's an abstraction on top of logical clusters that's going to properly bootstrap and prepare a logical cluster for the use case that you want.

100% this. And they keep everything alive. When you delete the workspace, the logical cluster goes away (through finalizer on the workspace). But you can create logical clusters that have no parent at all. The "root" workspace is one of them. But there can be many.

Reply:
> What I still don't understand is the point about CRUD endpoints. If that's what I get just with a logical cluster (which is my understanding of https://github.com/kcp-dev/kcp/pull/3238#discussion_r1898576991), what endpoints are actually provided by a Workspace?

You can talk to a logical cluster via its name, or alternatively through the workspace hierarchy path. In that sense, the workspace adds more CRUD endpoints, but I would rather call them aliases of those already existing from the logical cluster.

E.g. you can talk to a workspace through
1. `/clusters/2323kjklj234lk` – the logical cluster name
2. `/clusters/root:foo:bar` – the workspace path.
Both endpoints give you the same result.